sqlglot/dialects/snowflake.py

```diff
@@ -1376,3 +1376,25 @@ def select_sql(self, expression: exp.Select) -> str:
             if offset and not limit:
                 expression.limit(exp.Null(), copy=False)
             return super().select_sql(expression)
+
+        def createable_sql(self, expression: exp.Create, locations: t.DefaultDict) -> str:
+            kind = self.sql(expression, "kind").upper()
+            is_materialized = expression.find(exp.MaterializedProperty)
+            copy_grants_property = expression.find(exp.CopyGrantsProperty)
+
+            if kind == "VIEW" and is_materialized and copy_grants_property:
+                # For materialized views, COPY GRANTS is located *before* the columns list
+                # This is in contrast to normal views where COPY GRANTS is located *after* the columns list
+                # We default CopyGrantsProperty to POST_SCHEMA which means we need to output it POST_NAME if a materialized view is detected
+                # ref: https://docs.snowflake.com/en/sql-reference/sql/create-materialized-view#syntax
+                # ref: https://docs.snowflake.com/en/sql-reference/sql/create-view#syntax
+                post_schema_properties = locations.get(exp.Properties.Location.POST_SCHEMA, [])
```

Comment:
Current code can fail in the `index` call below if the lookup fails here:
```suggestion
                post_schema_properties = locations[exp.Properties.Location.POST_SCHEMA]
```

Otherwise, you could also slightly rewrite as follows if you want to be defensive:

```python
post_schema_properties = locations.get(exp.Properties.Location.POST_SCHEMA, [])
if copy_grants_property in post_schema_properties:
    post_schema_properties.remove(copy_grants_property)
```

But not sure why we'd fail to either find that property location in `locations` or fail to get the index of the property.